streams-components/streams-http/src/main/java/org/apache/streams/components/http/provider/SimpleHttpProvider.java

```diff
@@ -0,0 +1,324 @@
+
+            }
+        });
+    }
+
+    @Override
+    public StreamsResultSet readCurrent() {
+        StreamsResultSet current;
+
+        uriBuilder = uriBuilder.setPath(
+            Joiner.on("/").skipNulls().join(uriBuilder.getPath(), configuration.getResource(), configuration.getResourcePostfix())
+        );
+
+        URI uri;
+        try {
+            uri = uriBuilder.build();
+        } catch (URISyntaxException e) {
+            uri = null;
+        }
+
+        List<ObjectNode> results = execute(uri);
+
+        lock.writeLock().lock();
+
+        for( ObjectNode item : results ) {
+            providerQueue.add(newDatum(item));
+        }
+
+        LOGGER.debug("Creating new result set for {} items", providerQueue.size());
+        current = new StreamsResultSet(providerQueue);
+
+        return current;
+    }
+
+    protected List<ObjectNode> execute(URI uri) {
+
+        Preconditions.checkNotNull(uri);
+
+        List<ObjectNode> results = new ArrayList<>();
+
+        HttpRequestBase httpRequest = prepareHttpRequest(uri);
+
+        CloseableHttpResponse response = null;
+
+        String entityString = null;
+        try {
+            response = httpclient.execute(httpRequest);
+            HttpEntity entity = response.getEntity();
+            // TODO: handle retry
+            if (response.getStatusLine().getStatusCode() == 200 && entity != null) {
+                entityString = EntityUtils.toString(entity);
+                if( !entityString.equals("{}") && !entityString.equals("[]") ) {
+                    JsonNode jsonNode = mapper.readValue(entityString, JsonNode.class);
+                    results = parse(jsonNode);
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.error("IO error:\n{}\n{}\n{}", uri.toString(), response, e.getMessage());
+        } finally {
+            try {
+                response.close();
+            } catch (IOException e) {}
+        }
+        return results;
+    }
+
+    /**
+     Override this to change how entity gets converted to objects
+     */
+    protected List<ObjectNode> parse(JsonNode jsonNode) {
+
+        List<ObjectNode> results = new ArrayList<>();
+
+        if (jsonNode != null && jsonNode instanceof ObjectNode ) {
+            results.add((ObjectNode) jsonNode);
+        } else if (jsonNode != null && jsonNode instanceof ArrayNode) {
+            ArrayNode arrayNode = (ArrayNode) jsonNode;
+            Iterator<JsonNode> iterator = arrayNode.elements();
+            while (iterator.hasNext()) {
+                ObjectNode element = (ObjectNode) iterator.next();
+
+                results.add(element);
+            }
+        }
+
+        return results;
+    }
+
+    /**
+     Override this to change how metadata is derived from object
+     */
+    protected StreamsDatum newDatum(ObjectNode item) {
+        return new StreamsDatum(item, item.get("id").asText(), new DateTime(item.get("timestamp").asText()));
```

Comment:
I think it'd make sense to check for the presence of id and timestamp and use the different StreamsDatum constructors depending on their presence. This assumes a lot about the item as it currently is.